File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/DownKeyCommandPressUp.java
package com.example.try_gameengine.remotecontroller;

import android.view.MotionEvent;

import com.example.try_gameengine.remotecontroller.RemoteController.CommandType;

public class DownKeyCommandPressUp implements Command {
	DownKey downKey;
	private int pointerId = -1;

	public DownKeyCommandPressUp(DownKey downKey) {
		this.downKey = downKey;
	}

	@Override
	public CommandType execute() {
		return CommandType.DownKeyUpCommand;
	}

	@Override
	public boolean checkExecute(float x, float y, MotionEvent event) {
		return downKey.pressUp(x, y, event);
	}

	@Override
	public void setMotionEventPointerId(int pointerId) {
		this.pointerId = pointerId;
	}

	@Override
	public int getMotionEventPointerId() {
		return pointerId;
	}

}

File: Try_GameEngine5/src/com/example/try_gameengine/remotecontroller/DownKeyCommandPressUp.kt
package com.example.try_gameengine.remotecontroller

import android.view.MotionEvent

class DownKeyCommandPressUp(downKey: DownKey) : Command {
    var downKey: DownKey
    private var pointerId = -1

    init {
        this.downKey = downKey
    }

    override fun execute(): RemoteController.CommandType {
        return RemoteController.CommandType.DownKeyUpCommand
    }

    override fun checkExecute(x: Float, y: Float, event: MotionEvent?): Boolean {
        return downKey.pressUp(x, y, event)
    }

    override fun setMotionEventPointerId(pointerId: Int) {
        this.pointerId = pointerId
    }

    override fun getMotionEventPointerId(): Int {
        return pointerId
    }
}
